packages/editor/src/components/entities-saved-states/index.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * External dependencies
+ */
+import { startCase } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Button, CheckboxControl, Modal } from '@wordpress/components';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useState, useCallback } from '@wordpress/element';
+
+const EntitiesSavedStatesCheckbox = ( {
+	id,
+	name,
+	changes: { rawRecord },
+	checked,
+	setCheckedById,
+} ) => (
+	<CheckboxControl
+		label={ `${ startCase( name ) }: "${ rawRecord.title || rawRecord.name }"` }
+		checked={ checked }
+		onChange={ useCallback( ( nextChecked ) => setCheckedById( id, nextChecked ), [ id ] ) }
+	/>
+);
+
+export default function EntitiesSavedStates( { isOpen, onRequestClose } ) {
+	const entityRecordChangesByRecord = useSelect( ( select ) =>
+		select( 'core' ).getEntityRecordChangesByRecord()
+	);
+	const { saveEditedEntityRecord } = useDispatch( 'core' );
+
+	const [ checkedById, _setCheckedById ] = useState( {} );
+	const setCheckedById = useCallback(
+		( id, checked ) =>
+			_setCheckedById( ( prevCheckedById ) => {
+				const nextCheckedById = {
+					...prevCheckedById,
+				};
+				if ( checked ) {
+					nextCheckedById[ id ] = true;
+				} else {
+					delete nextCheckedById[ id ];
+				}
+				return nextCheckedById;
+			} ),
+		[]
+	);
+	const saveCheckedEntities = useCallback( () => {
+		Object.keys( checkedById ).forEach( ( id ) =>
+			saveEditedEntityRecord( ...id.split( ' | ' ).filter( ( s ) => s !== 'undefined' ) )
```

Comment:
I was hopeful that `undefined` would never be used.

I'll change it to `EquivalentKeyMap` to be sound.

Reply:
To clarify, we should use `EquivalentKeyMap`, because `kind` or `name` might contain the string `"undefined"` or `" | "` and break the logic. But, we still need to check and filter out the `id`/`key` if it equals the string `"undefined"`, because that's how base entities like root-site are stored in the store.

Reply:
> But, we still need to check and filter out the `id`/`key` if it equals the string `"undefined"`, because that's how base entities like root-site are stored in the store.

I guess it's not something we need to solve here, but that doesn't sound like an ideal way to structure the state. More broadly, I feel like the entire "kind", "name" pair system is pretty clunky in its usage, and would be worth revisiting.

Note: I'm just basing this off observation alone; I don't really know how viable (or the full extent of what would be involved in) refactoring this state.

Reply:
Agreed, unfortunately, this was already very ingrained in all of `core-data`.

We would need to move undefined-ID entities into their own slice of state instead of being nested inside `kind => name`. There would be a lot of changes involved.

I think it's safe to assume/enforce that IDs won't/don't `=== 'undefined'`.